h2o-bindings/bin/custom/python/gen_targetencoder.py

```diff
@@ -29,3 +43,134 @@ def transform(self, frame, data_leakage_handling="None", noise=-1, seed=-1):
+>>> predictors = ["home.dest", "cabin", "embarked"]
+>>> response = "survived"
+>>> titanic["survived"] = titanic["survived"].asfactor()
+>>> fold_col = "kfold_column"
+>>> titanic[fold_col] = titanic.kfold_column(n_folds=5, seed=1234)
+>>> titanic_te = H2OTargetEncoderEstimator(k = 0.7,
+...                                        f = 0.3,
+...                                        data_leakage_handling= "None",
+...                                        blending = True)
+>>> titanic_te.train(x = predictors,
+...                  y = response,
+...                  training_frame = titanic)
+>>> titanic_te
+""",
+    data_leakage_handling="""
+>>> titanic = h2o.import_file("https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv")
+>>> predictors = ["home.dest", "cabin", "embarked"]
+>>> response = "survived"
+>>> titanic["survived"] = titanic["survived"].asfactor()
+>>> fold_col = "kfold_column"
+>>> titanic[fold_col] = titanic.kfold_column(n_folds=5, seed=1234)
+>>> titanic_te = H2OTargetEncoderEstimator(k = 0.7,
+...                                        f = 0.3,
+...                                        data_leakage_handling= "k_fold",
+...                                        blending = True)
+>>> titanic_te.train(x = predictors,
+...                  y = response,
+...                  training_frame = titanic)
+>>> titanic_te
+""",
+    f="""
+>>> titanic = h2o.import_file("https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv")
+>>> predictors = ["home.dest", "cabin", "embarked"]
+>>> response = "survived"
+>>> titanic["survived"] = titanic["survived"].asfactor()
+>>> fold_col = "kfold_column"
+>>> titanic[fold_col] = titanic.kfold_column(n_folds=5, seed=1234)
+>>> titanic_te = H2OTargetEncoderEstimator(k = 0.7,
+...                                        f = 0.3,
+...                                        data_leakage_handling = "None",
+...                                        blending = True)
+>>> titanic_te.train(x = predictors,
+...                  y = response,
+...                  training_frame = titanic)
+>>> titanic_te
+""",
+    fold_column="""
+>>> titanic = h2o.import_file("https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv")
+>>> predictors = ["home.dest", "cabin", "embarked"]
+>>> response = "survived"
+>>> titanic["survived"] = titanic["survived"].asfactor()
+>>> fold_col = "kfold_column"
+>>> titanic[fold_col] = titanic.kfold_column(n_folds=5, seed=1234)
+>>> titanic_te = H2OTargetEncoderEstimator(k = 0.7,
+...                                        f = 0.3,
+...                                        data_leakage_handling = "None",
+...                                        blending = True)
+>>> titanic_te.train(x = predictors,
+...                  y = response,
+...                  training_frame = titanic)
+>>> titanic_te
+""",
+    ignored_columns="""
```

Comment:
please don't document this param (anywhere, I should say): code generator unfortunately exposes it but it's aimed at internal use only.